security/pkg/pki/ca/controller/secret.go

```diff
@@ -105,9 +106,13 @@ type SecretController struct {
 }
 
 // NewSecretController returns a pointer to a newly constructed SecretController instance.
-func NewSecretController(ca ca.CertificateAuthority, certTTL time.Duration, gracePeriodRatio float32, minGracePeriod time.Duration,
-	core corev1.CoreV1Interface, forCA bool, namespace string, dnsNames map[string]DNSNameEntry) (*SecretController, error) {
+func NewSecretController(ca ca.CertificateAuthority, certTTL time.Duration, identityDomain string,
+	gracePeriodRatio float32, minGracePeriod time.Duration, core corev1.CoreV1Interface, forCA bool,
+	namespace string, dnsNames map[string]DNSNameEntry) (*SecretController, error) {
 
+	if identityDomain == "" {
```

Comment:
For backward compatibility, shall we use default value `cluster.domain` if `identityDomain` was not specified?

Also as `identityDomain` always has a default value of `cluster.domain`, when this case will happen?

Reply:
Are you suggesting setting to `cluster.local` if unspecified here instead of an error?
I followed the same handling as invalid `gracePeriodRatio` - return an error instead of silently overriding on explicit parameters.

As an aside, the default was `cluster.local` (see previous version of README.md and code). This value is set as the default **if** the command line parameter is not specified.
Since the default value is only set via the commandline in security/istio_ca/cmd/main.go, direct calls to the secret controller can still contain an empty string (I expect these will be for testing mostly/only but left the guard in place anyway).

Reply:
> Since the default value is only set via the commandline in security/istio_ca/cmd/main.go, direct calls to the secret controller can still contain an empty string 

I think this is the reason why I want to set `cluster.local` as default value, as the end user may confused: Why it works in 1.0 but not work with version upper 1.0 without setting identityDomain? I think we need to keep the behavior unchanged.

Reply:
Changed to use the default if unspecified